vowpalwabbit/sample_pdf.cc

```diff
@@ -71,10 +71,10 @@ int sample_pdf::predict(example& ec, experimental::api_status*)
   return VW::experimental::error_code::success;
 }
 
-void sample_pdf::init(single_learner* p_base, uint64_t* p_random_seed)
+void sample_pdf::init(single_learner* p_base, std::shared_ptr<rand_state>&& random_state)
 {
   _base = p_base;
-  _p_random_state = p_random_seed;
+  _p_random_state = random_state;
```

Comment:
```suggestion
  _p_random_state = std::move(random_state);
```

Reply:
Do we want to move the shared pointer here?

Reply:
If the argument is by value then moving it here is what we want. If the arg is by ref then letting it copy here is good.

Reply:
yeah was wondering if it's safe but looks like all.`get_random_state()` returns a copy of the shared pointer, so passing it as an rvalue and moving it seems fine (and probably a tad faster)

Reply:
Either way works